Dockerfile

```diff
@@ -1,13 +1,22 @@
-FROM --platform=${BUILDPLATFORM:-linux/amd64} cgr.dev/chainguard/go:1.20 AS builder
+#syntax=docker/dockerfile:1.4
+
+FROM --platform=$BUILDPLATFORM tonistiigi/xx:1.1.0 AS xx
+
+FROM --platform=$BUILDPLATFORM golang:1.20-alpine AS builder
 ARG TARGETPLATFORM
 ARG BUILDPLATFORM
 ARG TARGETOS
 ARG TARGETARCH
 
+COPY --from=xx / /
```

Comment:
This was abit confusing for me not knowing this cross build platform: can we add a comment?

```dockerfile
# copy xx scripts to the builder stage
COPY --from=xx / /
```